### PR TITLE
Create QA view for Empty strings (fixes #813)

### LINKED ISSUE #813: Create QA view for empty strings
I noticed several locales doing strange things with empty strings, mostly to make tools happy.

I plan to create a view that shows empty strings in English and locale, now that we can differentiate them.

### PR DESCRIPTION
**Pull request by flodolo**

### CODE REVIEW

**Comment by TheoChevalier** [comment]
Getting these on central, but will check if it’s because locales are not up-to-date:

[Mon Sep 26 14:52:22 2016] PHP Notice:  Undefined index: toolkit/chrome/mozapps/help/help.dtd:fullZoomEnlargeCmd.commandkey3 in /Users/tchevalier/moz/transvision/app/models/empty_strings.php on line 47
[Mon Sep 26 14:52:22 2016] PHP Notice:  Undefined index: toolkit/chrome/mozapps/help/help.dtd:fullZoomReduceCmd.commandkey2 in /Users/tchevalier/moz/transvision/app/models/empty_strings.php on line 47
[Mon Sep 26 14:52:22 2016] PHP Notice:  Undefined index: toolkit/chrome/mozapps/help/help.dtd:fullZoomResetCmd.commandkey2 in /Users/tchevalier/moz/transvision/app/models/empty_strings.php on line 47

**Comment by flodolo** [comment]
Interesting. I didn't put a check on that line because we don't store obsolete strings, so a locale shouldn't be expected to have extra keys. I guess I can add the check like the previous cycle, just in case.

**Comment by TheoChevalier** [comment]
Looks like we need to escape html tags, they are interpreted at the bottom of http://localhost:8082/empty-strings/?locale=de&repo=central

**Comment by TheoChevalier** [comment]
(And the errors above were definitely due to l10n repos not being up-to-date)

File: app/models/empty_strings.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            'reference'   => $reference_strings[$string_id],
+            'translation' => $value,
+        ];
+    }
+}
+
+unset($reference_strings);
+unset($locale_strings);
+
+if (count($empty_strings > 0)) {
```
Comment (TheoChevalier) [inline app/models/empty_strings.php]: Hm, this does not seem to work, I get both the filter block and "No strings found" with /empty-strings/?locale=eo&repo=central

**Comment by TheoChevalier** [comment]
Maybe we should separate results into two groups: when the source is empty, and when the locale is empty? Or make the columns sortable?

**Comment by flodolo** [comment]
My first approach was to have 2 separate tables, but then I decided to only show the strings that diverge (i.e. not both empty), and given the small number of results it seemed OK to only have one table.

Added the missing .js to make them sortable

File: app/models/empty_strings.php (hunk 1 above)
Comment (flodolo) [inline app/models/empty_strings.php]: Yeah, if I only I were able to put the parenthesis where they belong… (fixing)

**Comment by TheoChevalier** [comment]
Thanks, looks good 👍

**Pull request merged**